src/container_runner.py

```diff
@@ -152,6 +156,23 @@ def __init__(self, container_image: str, container_port: int, host_port: int):
         self._container_port = container_port
         self._host_port = host_port
 
+    def set_docker_proxy(self, http_proxy: str, https_proxy: str):
```

Comment:
Like we discussed on our call, the logic here should be inlined into `_set_proxy` and the whole thing should be lifted out as a top level function as `self` isn't used anywhere.

That should also let you inline building out the proxy config which should reduce the line count a bit :slightly_smiling_face: